src/pages/TermsOfService.ts

```diff
@@ -0,0 +1,148 @@
+  const acceptedLatestTos = isSignedIn && termsOfService.userHasAcceptedLatestTos;
+  const usageAllowed = isSignedIn && termsOfService.permitsSystemUsage;
+  const [tosText, setTosText] = useState();
+
+  useOnMount(() => {
+    const loadTosAndUpdateState = _.flow(
+      Utils.withBusyState(setBusy),
+      withErrorReporting('There was an error retrieving our terms of service.')
+    )(async () => {
+      setTosText(await Ajax().User.getTos());
+    });
+    loadTosAndUpdateState();
+  });
+
+  const accept = async () => {
+    try {
+      setBusy(true);
+      const { enabled } = await Ajax().User.acceptTos();
+      const termsOfService = await Ajax().User.getTermsOfServiceComplianceStatus();
+
+      if (enabled) {
+        const registrationStatus = 'registered';
+        authStore.update((state) => ({ ...state, registrationStatus, termsOfService }));
+        Nav.goToPath('root');
+      } else {
+        reportError('Error accepting TOS, unexpected backend error occurred.', new Error('Cannot accept ToS'));
+      }
+    } catch (error) {
+      reportError('Error accepting Terms of Service', error);
+    } finally {
+      setBusy(false);
+    }
+  };
+
+  const reject = async () => {
+    try {
+      setBusy(true);
+      await Ajax().User.rejectTos();
+    } catch (error) {
+      reportError('Error rejecting Terms of Service', error);
+    } finally {
+      setBusy(false);
+      signOut('declinedTos');
+    }
+  };
+
+  const continueButton = () => {
+    Nav.goToPath('root');
+  };
+
+  return div(
+    {
+      role: 'main',
+      style: { padding: '1rem', minHeight: '100%', display: 'flex', justifyContent: 'center', alignItems: 'center' },
+    },
+    [
+      img({
+        src: scienceBackground,
+        alt: '',
+        style: { position: 'fixed', top: 0, left: 0, zIndex: -1 },
+      }),
+      div(
+        {
+          style: {
+            backgroundColor: 'white',
+            borderRadius: 5,
+            width: 800,
+            maxHeight: '100%',
+            padding: '2rem',
+            boxShadow: Style.standardShadow,
+          },
+        },
+        [
+          h1({ style: { color: colors.dark(), fontSize: 38, fontWeight: 400 } }, ['Terra Terms of Service']),
+          !acceptedLatestTos &&
+            div({ style: { fontSize: 18, fontWeight: 600 } }, ['Please accept the Terms of Service to continue.']),
+          div(
+            { style: { height: '50vh', overflowY: 'auto', lineHeight: 1.5, marginTop: '1rem', paddingRight: '1rem' } },
+            [
+              !tosText
+                ? centeredSpinner()
+                : h(
+                    MarkdownViewer,
+                    {
+                      renderers: {
+                        link: newWindowLinkRenderer,
+                        heading: (text, level) => `<h${level} style="margin-bottom: 0">${text}</h${level}>`,
+                      },
+                    } as WithKey<any>,
```

Comment:
The error here is the result of incorrect type definitions in JSDoc on MarkdownViewer. Instead of casting, let's use `@ts-expect-error` here so that we're notified to remove the workaround when MarkdownViewer is fixed.

Reply:
Ok that's good to know. I was very confused trying to get this to accept the types.